pandas/tests/frame/test_missing.py

```diff
@@ -671,6 +671,21 @@ def test_fillna_columns(self):
         expected = df.astype(float).fillna(method="ffill", axis=1)
         assert_frame_equal(result, expected)
 
+    def test_fillna_rows(self):
+        #GH17399
+        df = pd.DataFrame({
+            "a": [1,2,3,4],
+            "b": [5,np.nan,7,8],
+            "c": [9,10,11,np.nan]})
+
+        expected = pd.DataFrame({
+            "a": [1,2,3,4],
+            "b": [5,6,7,8],
+            "c": [9,10,11,6]})
+
+        result = df.fillna(df.mean(axis=1))
```

Comment:
I don't see `axis` passed to fillna here? You pass it to mean but not fillna.

Reply:
Maybe we should back up and make sure the behavior I'm seeing is expected then an not a bug.  In [this notebook](https://imgur.com/a/3ZrXqSv) I first try to replace NaNs with the column average, then the row average.  Doing the first works, but the second silently does nothing. 

Setting axis=1 on fillna when using the mean where axis=1 causes an error:  Currently only can fill with dict/Series column by column.  

It seems like calling fillna on a frame with NaNs which doesn't fill them is an issue.